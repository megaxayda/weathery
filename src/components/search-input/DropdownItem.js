import React, { memo } from 'react';
import Dropdown from 'react-bootstrap/Dropdown';
import PropTypes from 'prop-types';

function DropdownItem({ title, value, onSelect }) {
  const _onSelect = () => {
    onSelect(title, value);
  };

  return <Dropdown.Item onClick={_onSelect}>{title}</Dropdown.Item>;
}

export default memo(DropdownItem);

DropdownItem.propTypes = {
  title: PropTypes.string.isRequired,
  value: PropTypes.number.isRequired,
  onSelect: PropTypes.func.isRequired,
};
