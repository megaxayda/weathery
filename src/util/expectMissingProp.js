import sinon from 'sinon';

function expectMissingProp(prop, component) {
  sinon.assert.calledWithMatch(
    console.error,
    new RegExp(`Warning: Failed prop type: The prop \`${prop}\` is marked as required in \`${component}\`, but its value is \`undefined\`.`)
  );
}

export default expectMissingProp;
