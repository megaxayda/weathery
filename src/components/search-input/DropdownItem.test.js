import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import PropTypes from 'prop-types';
import sinon from 'sinon';
import renderer from 'react-test-renderer';

import expectMissingProp from 'util/expectMissingProp';
import DropdownItem from './DropdownItem';

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement('div');
  document.body.appendChild(container);
  sinon.stub(console, 'error');
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
  console.error.restore();
  PropTypes.resetWarningCache();
});

test('check props type', () => {
  act(() => {
    render(<DropdownItem />, container);
  });

  sinon.assert.callCount(console.error, 3);
  expectMissingProp('title', 'DropdownItem');
  expectMissingProp('value', 'DropdownItem');
  expectMissingProp('onSelect', 'DropdownItem');
});

test('render DropdownItem', () => {
  const tree = renderer.create(<DropdownItem title={'Tokyo'} value={123} onSelect={() => {}} />).toJSON();
  expect(tree).toMatchSnapshot();
});
