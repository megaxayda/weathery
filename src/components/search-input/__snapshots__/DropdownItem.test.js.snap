// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render DropdownItem 1`] = `
<a
  className="dropdown-item"
  href="#"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="button"
>
  Tokyo
</a>
`;
